packages/ember-template-compiler/lib/plugins/transform-component-invocation.ts

```diff
@@ -172,21 +172,13 @@ function isInlineInvocation(
 }
 
 function isIllegalName(node: AST.PathExpression, hasLocal: (k: string) => boolean): boolean {
-  return isThisPath(node) || isDotPath(node) || isNamedArg(node) || isLocalVariable(node, hasLocal);
```

Comment:
Just to confirm, removing `isThisPath` here because you are now allowed to have this paths in the VM, right?